src/OVAL/oval_component.c

```diff
@@ -1973,6 +2051,159 @@ static oval_syschar_collection_flag_t _oval_component_evaluate_ESCAPE_REGEX(oval
 	return flag;
 }
 
+#define INITIAL_CAPACITY 64
+
+typedef enum {
+    START,
+    LEFT_BRACKET,
+    CLASS,
+    ESCAPE
+} states;
+
+typedef struct string {
+    char *str;
+    unsigned int length;
+    unsigned int capacity;
+} t_string;
+
+static t_string *_string_new() {
+    t_string *s;
+    s = malloc(sizeof(t_string));
+    if (s == NULL)
+        return NULL;
+    s->str = malloc(INITIAL_CAPACITY);
+    if (s->str == NULL) {
+        free(s);
+        return NULL;
+    }
+    s->str[0] = '\0';
+    s->length = 0;
+    s->capacity = INITIAL_CAPACITY;
+    return s;
+}
+
+static void _string_append(t_string *s, char c)
+{
+    if (s->length + 1 >= s->capacity) {
+        if(realloc(s->str, s->capacity + INITIAL_CAPACITY) == NULL)
+            return;
+        s->capacity += INITIAL_CAPACITY;
+    }
+    s->str[s->length++] = c;
+    s->str[s->length] = '\0';
+}
+
+static char *_glob_to_regex (const char *glob, int noescape)
```

Comment:
Discussed with @jan-cerny. Found cases where _glob_to_regex reads invalid memory. jcerny is working on a fix.